api/src/main/java/org/openmrs/module/registrationcore/api/impl/RegistrationCoreServiceImpl.java

```diff
@@ -367,27 +368,26 @@ public boolean supportsPropertyName(String gpName) {
 	}
 
 	@Override
-	public String importMpiPatient(String personId) {
-		if (registrationCoreProperties.isMpiEnabled()) {
-			MpiProvider mpiProvider = registrationCoreProperties.getMpiProvider();
-			Patient importedPatient = mpiProvider.fetchMpiPatient(personId);
-			return persistImportedMpiPatient(importedPatient);
-		} else {
-			//should not pass here since "importPatient" performs only when MpiProvider is not null
-			throw new APIException("Should not perform 'fetchMpiPatient' when MPI is disabled");
+	public Patient findMpiPatient(String identifier, String identifierTypeUuid) {
+		if (!registrationCoreProperties.isMpiEnabled()) {
+			throw new MpiException("Should not perform 'findMpiPatient' when MPI is disabled");
 		}
+		MpiProvider mpiProvider = registrationCoreProperties.getMpiProvider();
+		return mpiProvider.fetchMpiPatient(identifier, identifierTypeUuid);
+	}
+
+	@Override
+	public String importMpiPatient(String personId) {
+		PatientIdentifierType patientIdentifierType = patientService.getPatientIdentifierTypeByName(
```

Comment:
I understand that there is no need to check if the mpi support is enabled here because it will be verified in findMpiPatient. However, I'm not if the error message returned by an exception will be accurate (""Should not perform 'findMpiPatient' when MPI is disabled")

Reply:
Hmm, I think that it could stay. Finding MPI patient is closely related to importing them.